### DISCUSSION TITLE
GPTAssistantAgent(Coder Assistant) does not want to terminate

### DISCUSSION THREAD

**Discussion by shaleenrajkaur** [microsoft/autogen#2085] [Q&A]
### Describe the issue

Hello! I've using GPTAssistant to interact with the userpoxy agent of Autogen and in the message the response says terminate

```
the average closing price in this period is approximately $457.27.

TERMINATE
```

but the code still RUNS do we know how to solve this

**Comment by ekzhu** [comment]
In this case you can set the `is_termination_msg` lambda function in `UserProxyAgent`'s constructor to detect if the "content" field of the message ends or contains the word "TERMINATE".

**Reply by shaleenrajkaur** [reply]
def is_termination_msg(data):
    has_content = "content" in data and data["content"] is not None
    return has_content and "TERMINATE" in data["content"]

file = client.files.create(
    file=open("/Users/shaleenraj_kaur/Downloads/20230312 stock_data.csv", "rb"),
    purpose='assistants'
)

assistant = client.beta.assistants.create(
    name="Python Developer",
    instructions="You are a python developer",
    model="gpt-4-1106-preview",
    tools = [ { "type": "code_interpreter" } ],
    file_ids=[file.id]
)

config_list = [{"model": "gpt-4", 'api_key': "skx"}]


llm_config = { 
    "assistant_id": assistant.id
}

gpt_assistant = GPTAssistantAgent(
    name="Coder Assistant",
    instructions="""
    You are an expert at writing python code to solve problems. 
    Upon successful completion of the task , add "TERMINATE" to conclude the message.
    """,
    llm_config=llm_config,
    is_termination_msg=is_termination_msg,
)

user_proxy = autogen.UserProxyAgent(
    name="shAI",
    code_execution_config={
"work_dir": "coding",
"use_docker": False, # set to True or image name like "python:3" to use docker
}
)




user_proxy.initiate_chat(
    gpt_assistant,
    message="""
   What is the average closing price in this period?
    """,
    human_input_mode="NEVER",
    max_consecutive_auto_reply=1,
    is_termination_msg=is_termination_msg

)
    
    
    I have used that would you recommend anything else?

**Reply by sonichi** [reply]
Please pass `is_termination_msg` to the `UserProxyAgent` constructor.
Also `human_input_mode`, `max_consecutive_auto_reply`
